src/auth/manager/resolver.go

```diff
@@ -0,0 +1,59 @@
+package manager
+
+import (
+	"fmt"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+)
+
+type Result struct {
+	isAllowed bool
+	err       error
+}
+
+func (res *Result) Allowed() bool {
+	return res.isAllowed
+}
+
+func (res *Result) Error() error {
+	return res.err
+}
+
+// Resolver is responsible to control whether an operation is authorized or not
+// depending on the set of policies attached to the resolver
+type Resolver struct {
```

Comment:
I don't see the need for a `Resolver` at all. It seems that we can simply do a `permissions.Contains("read:keys")` on the connector and it should be enough right? What I mean is that we can use pure functions and do not need a stateful struct.

Reply:
You are right at the current version having a Resolver appers to be not very necesiary but IMO this dedicated component help us to decouple out authorization logic out from the Connectors and improve scalability in case we decide on making more advance auth rules such as ownership base rules where the Resolver would need to read from DB data.